bccsp/sw/fileks.go

```diff
@@ -65,32 +66,51 @@ func (ks *fileBasedKeyStore) Init(pwd []byte, path string, readOnly bool) error
 	// pwd can be nil
 
 	if len(path) == 0 {
-		return errors.New("An invalid KeyStore path provided. Path cannot be an empty string.")
+		return errors.New("an invalid KeyStore path provided. Path cannot be an empty string")
 	}
 
 	ks.m.Lock()
 	defer ks.m.Unlock()
 
 	if ks.isOpen {
-		return errors.New("KeyStore already initilized.")
+		return errors.New("keystore is already initialized")
 	}
 
 	ks.path = path
-	ks.pwd = utils.Clone(pwd)
 
-	err := ks.createKeyStoreIfNotExists()
+	clone := make([]byte, len(pwd))
+	copy(ks.pwd, pwd)
+	ks.pwd = clone
+	ks.readOnly = readOnly
+
+	// Check if the keystore directory exists
+	exists, err := dirExists(path)
 	if err != nil {
 		return err
 	}
+	// If keystore directory does not exist, create the keystore
```

Comment:
Nit. You have refactored the code in a nice way and It is very clear from the code that it checks whether the dirExists and if not, create the store. IMO, this comment and the one at line 86 does not help. While reading the code, it creates an unnecessary pause and a breakage. Please refer to this PR for discussion on comments -- [895](https://github.com/hyperledger/fabric/pull/895). The same suggestion is applicable to dirEmpty related code below. 

Sorry If you think I am doing too much nitpicking. It is just that we can reduce 4 lines of code and avoid breakage during the code reading. If you think this will improve the code readability, please follow this rule in other future PRs too. IMO, these simple things also improve the code readability. 

Throughout the file, you can notice comments like (exists already, not added by you)
```
// Load the key
key, err := ks.loadKey(hex.EncodeToString(ski))
```
```
// Load the private key
key, err := ks.loadPrivateKey(hex.EncodeToString(ski))
```
```
// Load the public key
key, err := ks.loadPublicKey(hex.EncodeToString(ski))
```
If the golang speaks greek langauge instead of English, I think it is okay to have this English translation :-)

Reply:
Agreed and removed.

And while it is nitpicking, we have to start somewhere. The more we educate everyone on best practices, and standardizing on those practices, hopefully the more good practices will spread throughout the codebase. 

Thanks Senthil!